### PR TITLE
Consolidate paragraph translation into host translation requests.

### PR DESCRIPTION
**Pull request by Etroid**
## Release Notes
- [ ] Refactors `loadEntity` function to ensure the correct "latest translation affected" revision is loaded when no revision id is provided.
- [ ] Adds support for paragraph translation as part of host entity translation requests.

### CODE REVIEW

File: src/ContentEntityTranslate.php [hunk 1]
```diff
@@ -95,6 +97,46 @@ public function translate(ContentEntityInterface $srcEntity, array $translation)
...
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\TypedData\Exception\MissingDataException
+   * @throws \Drupal\Core\TypedData\Exception\ReadOnlyException
+   * @throws \Drupal\typed_data\Exception\InvalidArgumentException
+   */
+  public function translateParagraphs(ContentEntityInterface $targetEntity, $targetLangcode, array $translations) {
+    foreach ($targetEntity->getFields() as $fieldItemList) {
+      if ($fieldItemList instanceof EntityReferenceFieldItemListInterface && $fieldItemList->getSetting('target_type') ==='paragraph') {
```
Comment (angrytoast) [inline src/ContentEntityTranslate.php]: minor nit, missing space after the `===`

File: src/ContentEntityTranslate.php [hunk 2]
```diff
@@ -95,6 +97,46 @@ public function translate(ContentEntityInterface $srcEntity, array $translation)
...
+              if ($uuid === $translation['entityId']) {
+                $paragraph = $this->translate($paragraph->getTranslation($targetLangcode), $targetLangcode, $translation);
+                $this->translateParagraphs($paragraph, $targetLangcode, $translations);
+                $paragraph->setNewRevision(TRUE);
+                $paragraph->setNeedsSave(TRUE);
+                break;
+              }
+            }
+
+            $fieldItemList[$delta] = $paragraph;
```
Comment (angrytoast) [inline src/ContentEntityTranslate.php:133]: Q: What does overriding the `$fieldItemList[$delta]` accomplish here?

**Review by angrytoast — COMMENTED** [review]
One minor nit and a question on a code point.

File: src/ContentEntityTranslate.php (hunk 2 above)
Comment (Etroid) [reply src/ContentEntityTranslate.php:133]: Basically re-attaching the translated paragraph to the host node paragraphs field.

Comment (Etroid) [reply src/ContentEntityTranslate.php:133]: Another way of writing could be `$item->entity = $paragraph` , but I followed the pattern of the paragraph_asymmetric_translation module

Comment (angrytoast) [reply src/ContentEntityTranslate.php:133]: Got it 👍

**Comment by Etroid** [comment]
@rhf104 this is ready to be merged. If you approve can we roll a new release version. I'll send a separate e-mail to talk about discussing deployment times.

**Review by rhf104 — APPROVED** [review]
Looks good! I'll create a new Argo module release and open a PR on www.

**Pull request merged**